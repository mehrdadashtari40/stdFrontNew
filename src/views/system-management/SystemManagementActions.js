import getAuthonticatedJSON from '../../common/utils/functions/getAuthenticatedJSON';

export const FETCH_CATEGORIES = 'FETCH_CATEGORIES';
export const FETCH_CATEGORY = 'FETCH_CATEGORY';
export const LOAD_USER = 'LOAD_USER';
export const LOAD_GROUPS = 'LOAD_GROUPS';
export const LOAD_DEPARTMENTS = 'LOAD_DEPARTMENTS';
export const LOAD_DEPARTMENT = 'LOAD_DEPARTMENT';



export function loadCategories(url) {
	return dispatch => {
		return getAuthonticatedJSON(url).then(response => {
			dispatch(setCategories(response));
		});
	};
}

export function setCategories(data) {
	return {
		type: FETCH_CATEGORIES,
		Categories: data,
	};
}

export function loadCategory(url) {
	return dispatch => {
		return getAuthonticatedJSON(url).then(response => {
			dispatch(setCategory(response));
		});
	};
}

export function setCategory(data) {
	return {
		type: FETCH_CATEGORY,
		Category: data,
	};
}

export function loadUser(url) {
	return dispatch => {
		return getAuthonticatedJSON(url).then(response => {
			dispatch(setUser(response));
		});
	};
}

export function setUser(data) {
	return {
		type: LOAD_USER,
		User: data,
	};
}

export function loadGroups(url) {
	return dispatch => {
		return getAuthonticatedJSON(url).then(response => {
			dispatch(setGroups(response));
		});
	};
}

export function setGroups(data) {
	return {
		type: LOAD_GROUPS,
		Groups: data,
	};
}

export function loadDepartments(url) {
	return dispatch => {
		return getAuthonticatedJSON(url).then(response => {
			dispatch(setDepartments(response));
		});
	};
}

export function setDepartments(data) {
	return {
		type: LOAD_DEPARTMENTS,
		Departments: data,
	};
}


export function loadDepartment(url) {
	return dispatch => {
		return getAuthonticatedJSON(url).then(response => {
			dispatch(setDepartment(response));
		});
	};
}

export function setDepartment(data) {
	return {
		type: LOAD_DEPARTMENT,
		Department: data,
	};
}


