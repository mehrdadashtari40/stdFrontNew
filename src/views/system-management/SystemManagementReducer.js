import {
	FETCH_CATEGORIES,
	FETCH_CATEGORY,
	LOAD_USER,
	LOAD_GROUPS,
	LOAD_DEPARTMENTS,
	LOAD_DEPARTMENT,
} from './SystemManagementActions';

const initialState = {
	Categories: [],
	Category: [],
	User: [],
	Users: [],
	Groups: [],
	Departments: [],
	Department: [],
};

export function systemManagementReducer(state = initialState, action) {
	switch (action.type) {
		case FETCH_CATEGORIES:
			return {
				...state,
				Categories: action.Categories,
			};

		case FETCH_CATEGORY:
			return {
				...state,
				Category: action.Category,
			};

		case LOAD_USER:
			return {
				...state,
				User: action.User,
			};
		case LOAD_GROUPS:
			return {
				...state,
				Groups: action.Groups,
			};
		case LOAD_DEPARTMENTS:
			return {
				...state,
				Departments: action.Departments,
			};
		case LOAD_DEPARTMENT:
			return {
				...state,
				Department: action.Department,
			};

		default:
			return state;
	}
}
