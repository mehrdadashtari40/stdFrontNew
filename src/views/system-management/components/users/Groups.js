import React from 'react';
import $ from 'jquery';
import getAuthonticatedJSON from '../../../../common/utils/functions/getAuthenticatedJSON';
import {config} from '../../../../config/config';
import {smallBox, SmartMessageBox} from '../../../../common';
import Form from '../../utils/GroupForm';
import DualListBox from 'react-dual-listbox';
import {connect} from 'react-redux';
import * as Actions from '../../SystemManagementActions';
import {withTranslation} from 'react-i18next';
import {compose} from 'redux';
import MaterialDataTable from "../../../../common/tables/components/MaterialDataTable";
import postAuthenticatedJSON from "../../../../common/utils/functions/postAuthenticatedJSON";
import deleteAuthonticatedJSON from "../../../../common/utils/functions/deleteAuthonticatedJSON";

class Groups extends React.Component {
    constructor(props) {
        super(props);
        this.state = {
            Group: [],
            Edit: false,
            Show: [],
            refresh: false,
            GroupsMembers: [],
            GroupsMembersIds: [],
            GroupsMembersIdsConst: [],
            AssignUsers: [],
            AbaibleUsersIds: [],
            selectedForAssign: [],
            getUsesr: false,
            current_group_uid:null
        };
        this.handelIndex = this.handelIndex.bind(this);
        this.handelCreate = this.handelCreate.bind(this);
        this.handelShow = this.handelShow.bind(this);
        this.setGroupsMembers = this.setGroupsMembers.bind(this);
        this.getAvailableUsers = this.getAvailableUsers.bind(this);
        this.handelAssignUsers = this.handelAssignUsers.bind(this);
        this.assignUser = this.assignUser.bind(this);
        this.unAssignUser = this.unAssignUser.bind(this);
    }

    componentDidMount() {
        let self = this;
        this.handelIndex();
        $('#GroupBox').on('hidden.bs.modal', function () {
            self.handelIndex();
            // document.getElementById('new-role-form').reset();
        });
        $('#AssignUser').change(function () {
            alert();
        });
    }

    handelIndex() {
        this.props.loadGroups(config.apiServer + 'groups');
        this.setState({Edit: false});
    }

    handelCreate() {
        $('input[name=grp_uid]').val('');
        $('input[name=grp_title]').val('');
        $('select[name=grp_status]').val('');
        $('#actions').hide();
        $('#GroupBox').modal('show');
        this.enableFields();
    }

    handelAssignUsers(selected) {
        let forAssign = selected.filter(data => {
            return this.state.GroupsMembersIdsConst.indexOf(data) < 0;
        });
        let forUnAssign = this.state.GroupsMembersIdsConst.filter(data => {
            return selected.indexOf(data) < 0;
        });
        this.setState({GroupsMembersIds: selected});
        if (forAssign.length > 0) this.assignUser(forAssign);
        if (forUnAssign.length > 0) this.unAssignUser(forUnAssign);
    }

    handelShow(data) {
        getAuthonticatedJSON(config.apiServer + 'group/' + data['grp_uid'], {
            headers: {
                Authorization: 'Bearer ' + localStorage.getItem('access_token'),
                'Content-Type': 'application/json',
            },
        }).then(res => {
            $('#actions').show();
            this.setState({Show: res});
            $('#GroupBox').modal('show');
            this.SetData();
            this.disableFields();
            this.setState({
                current_group_uid: data['grp_uid']
            })
        });
    }

    SetData() {
        let data = this.state.Show;
        $('input[name=grp_uid]').val(data['grp_uid']);
        $('input[name=grp_title]').val(data['grp_title']);
        $('select[name=grp_status]').val(data['grp_status']);
    }

    enableFields() {
        $('select[name=grp_status]').prop('disabled', false);
        $('input[name=grp_title]').prop('disabled', false);
        return false;
    }

    disableFields() {
        $('select[name=grp_status]').prop('disabled', true);
        $('input[name=grp_title]').prop('disabled', true);
        return false;
    }

    assignUser(users) {
        let grpId = $('#grp_uid').val();
        postAuthenticatedJSON(config.apiServer + 'group/batch-users', [
            {
                groupUid: grpId,
                users: users,
            }
        ]).then(res => {
            smallBox({
                title: this.props.t('Successfully'),
                content: '<i>' + this.props.t('Data has been saved successfully') + '</i>',
                color: '#659265',
                iconSmall: 'fa fa-check fa-2x fadeInRight animated',
                timeout: 4000,
            });
        });
    }

    unAssignUser(unassigned) {
        let grpId = $('#grp_uid').val();
        try {
            for (let i = 0; i < unassigned.length; i++) {
                let usr_uid = unassigned[i];
                deleteAuthonticatedJSON(config.apiServer + 'group/' + grpId + '/user/' + usr_uid).then(res => {
                });
            }
            smallBox({
                title: this.props.t('Successfully'),
                content: '<i>' + this.props.t('Data has been saved successfully') + '</i>',
                color: '#659265',
                iconSmall: 'fa fa-check fa-2x fadeInRight animated',
                timeout: 4000,
            });
        } catch (e) {
        }
    }

    setGroupsMembers(data) {
        let a = [];
        data.map(d => {
            a.push(d.usr_uid);
        });
        this.setState({GroupsMembersIds: a});
        this.setState({GroupsMembersIdsConst: a});
        this.setState({GroupsMembers: data});

        $('#GroupMembers').modal('show');
    }

    handelDelete() {
        SmartMessageBox(
            {
                title: this.props.t('Delete confirm title'),
                content: this.props.t('Delete confirm content message'),
                buttons: '[' + this.props.t('No') + '][' + this.props.t('Yes') + ']',
            },
            ButtonPressed => {
                if (ButtonPressed === this.props.t('Yes')) {
                    let grpId = $('#grp_uid').val();
                    deleteAuthonticatedJSON(config.apiServer + 'group/' + grpId).then(res => {
                        smallBox({
                            title: this.props.t('Successfully'),
                            content: '<i>' + this.props.t('Data has been saved successfully') + '</i>',
                            color: '#659265',
                            iconSmall: 'fa fa-check fa-2x fadeInRight animated',
                            timeout: 4000,
                        });
                        $('#GroupBox').modal('hide');
                    });
                }
            }
        );
    }

    getAvailableUsers() {
        let grpId = $('#grp_uid').val();
        let self = this;
        getAuthonticatedJSON(config.apiServer + 'group/' + grpId + '/available-users', {
            headers: {
                Authorization: 'Bearer ' + localStorage.getItem('access_token'),
                'Content-Type': 'application/json',
            },
        }).then(res => {
            let AssignUsers = [];
            let AssignUsersids = [];
            res.map(item => {
                AssignUsers.push({
                    value: item.usr_uid,
                    label: '(' + item.usr_username + ')' + ' ' + item.usr_firstname + ' ' + item.usr_lastname,
                });
                AssignUsersids.push(item.usr_uid);
                return item;
            });
            self.state.GroupsMembers.map(item => {
                AssignUsers.push({
                    value: item.usr_uid,
                    label: '(' + item.usr_username + ') ' + item.usr_firstname + ' ' + item.usr_lastname,
                });
                return item;
            });
            self.setState({AssignUsers: AssignUsers});
            self.setState({AbaibleUsersIds: AssignUsersids});
            $('#AssignUsers').modal('show');
        });
    }

    closeAssignUserModal = () => {
        $('#AssignUsers').modal('hide');
        let status = !this.state.getUsesr;
        this.setState({getUsesr: status});
    };

    render() {
        let self = this;
        return (<><MaterialDataTable
            title={'گروه'}
            icon={'fas fa-table'}
            items={this.props.Groups}
            columns={[
                {
                    dataField: 'grp_title',
                    text: 'عنوان گروه',
                    filterable: 'grp_title'
                },
                {
                    dataField: 'grp_status',
                    text: 'وضعیت',
                    filterable: 'grp_status',
                    filterOptions: [{
                        key: 'active',
                        text: 'فعال'
                    }, {
                        key: 'inactive',
                        text: 'غیر فعال'
                    }
                    ],
                    formatter: function (cell, item) {
                        return item.grp_status === 'ACTIVE' ? <span className='label label-success'>فعال</span> :
                            <span className='label label-danger'>غیر فعال</span>
                    }
                },
                {
                    dataField: 'grp_users',
                    filterable: 'grp_users',
                    text: 'تعداد کاربران',
                },
                {
                    dataField: 'grp_tasks',
                    filterable: 'grp_tasks',
                    text: 'تعداد وظیفه',
                },
                {
                    dataField: 'actions',
                    text: 'عملیات',
                    formatter: function (cell, item) {
                        return (<div className={'table-actions'}>
                            <i className={'fas fa-edit'}
                               onClick={() => {
                                   self.handelShow(item)
                               }}
                            ></i>
                        </div>)
                    }
                },
            ]}
            getDataList={data => this.getUsers(data)}
            toolbar_btns={<>
                <button className="btn btn_inbox"
                        onClick={this.handelCreate}
                >
                    <i className="fa fa-plus" />
                    <span>گروه جدید</span>
                </button>
            </>}
        />

            <div id="GroupBox" className="modal fade" role="dialog">
                <div className="modal-dialog">
                    <div className="modal-content">
                        <div className="modal-header">
                            <button type="button" className="close" data-dismiss="modal">
                                &times;
                            </button>
                            <h4 className="modal-title"> {this.props.t('Group Details')}</h4>
                        </div>
                        <div className="modal-body">
                            <Form
                                Refresh={this.handelIndex}
                                Edit={this.state.Edit}
                                Disable={this.disableFields}
                                Enable={this.enableFields}
                                Data={this.state.Show}
                                setGroupsMembers={this.setGroupsMembers}
                                getUsers={this.state.getUsesr}
                            />
                        </div>
                    </div>
                </div>
            </div>
            <div id="GroupMembers" className="modal fade" role="dialog">
                <div className="modal-dialog modal-lg">
                    <div className="modal-content">
                        <div className="modal-header">
                            <button type="button" className="close" data-dismiss="modal">
                                &times;
                            </button>
                            <h4 className="modal-title"> {this.props.t('Users List')}</h4>
                        </div>
                        <div className="modal_option_bar padd_for_modal_optn">
                            <a className="btn btn-primary" onClick={this.getAvailableUsers}>
                                <i className="fa fa-users"></i>
                                {this.props.t('Assign Users')}
                            </a>
                        </div>
                        <div>
                            <MaterialDataTable
                                title={'کاربران'}
                                icon={'fas fa-table'}
                                url={config.apiServer + 'group/'+this.state.current_group_uid+'/users'}
                                fixedCount={5}
                                columns={[
                                    {
                                        dataField: 'usr_username',
                                        text: 'نام کاربری',
                                    },
                                    {
                                        dataField: 'usr_firstname',
                                        text: 'نام و نام خانوادگی',
                                        formatter: function (cell, item) {
                                            return item.usr_firstname + " " + item.usr_lastname
                                        }
                                    },
                                    {
                                        dataField: 'usr_status',
                                        text: 'وضعیت',
                                        formatter: function (cell, item) {
                                            return item.usr_status.toString() === 'ACTIVE' ? <span className='label label-success'>فعال</span> :
                                                <span className='label label-danger'>غیر فعال</span>
                                        }
                                    }
                                ]}
                                getDataList={data => this.getUsers(data)}
                            />
                        </div>
                        <div className="modal-footer">
                            <button type="button" className="btn btn-danger pull-right" data-dismiss="modal">
                                <i className="fa fa-close"></i> {this.props.t('Close')}
                            </button>
                        </div>
                    </div>
                </div>
            </div>
            <div id="AssignUsers" className="modal fade" role="dialog">
                <div className="modal-dialog modal-lg">
                    <div className="modal-content">
                        <div className="modal-header">
                            {/*<button type="button" className="close" data-dismiss="modal">&times;</button>*/}
                            <button type="button" className="close" onClick={this.closeAssignUserModal}>
                                &times;
                            </button>
                            <h4 className="modal-title">{this.props.t('Users List')}</h4>
                        </div>
                        <div>
                            {this.state.AssignUsers.length > 0 ? (
                                <DualListBox
                                    canFilter
                                    filterCallback={(option, filterInput) => {
                                        if (filterInput === '') {
                                            return true;
                                        }

                                        return new RegExp(filterInput, 'i').test(option.label);
                                    }}
                                    filterPlaceholder={this.props.t('Search')}
                                    options={this.state.AssignUsers}
                                    selected={this.state.GroupsMembersIds}
                                    onChange={this.handelAssignUsers}
                                />
                            ) : (
                                ''
                            )}
                        </div>
                        <div className="modal-footer">
                            <button
                                type="button btn btn-success"
                                className="btn btn-default"
                                data-dismiss="modal"
                                // onClick={this.assignUser}
                            >
                                {this.props.t('Submit')}
                            </button>
                        </div>
                    </div>
                </div>
            </div>
        </>);
    }
}

export default compose(withTranslation(), connect(state => state.systemManagement, Actions))(Groups);
