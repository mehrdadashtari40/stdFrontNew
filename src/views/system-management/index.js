import Categories from './components/setting/Categories';
import PmTables from './components/setting/PmTables';
import Backup from './components/backup/Backup';
import Setting from './components/setting/Setting';
import Users from './components/users/Users';
import Groups from './components/users/Groups';
import Department from './components/users/Departments';
export const routes = [
	{
		path: '/process-categories',
		exact: true,
		component: Categories,
		name: 'Categories',
	},
	{
		path: '/pm-tables',
		exact: true,
		component: PmTables,
		name: 'PmTables',
	},
	{
		path: '/backup',
		exact: true,
		component: Backup,
		name: 'Backup',
	},
	{
		path: '/setting',
		exact: true,
		component: Setting,
		name: 'Setting',
	},
	{
		path: '/users',
		exact: true,
		component: Users,
		name: 'Users',
	},
	{
		path: '/groups',
		exact: true,
		component: Groups,
		name: 'Groups',
  },
  
  {
		path: '/departments',
		exact: true,
		component: Department,
		name: 'Department',
	},
];
export * from './SystemManagementActions';
export * from './SystemManagementReducer';
