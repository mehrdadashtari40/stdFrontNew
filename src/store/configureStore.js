import { createStore, combineReducers, applyMiddleware } from 'redux';
import thunk from 'redux-thunk';

import { composeWithDevTools } from 'redux-devtools-extension';

import { userReducer, requestUserInfo } from '../common/user';
import { i18nReducer, i18nInit } from '../common/i18n';
import { trelloReducer } from '../views/trello';
import { systemManagementReducer } from '../views/system-management';
import { ProcessManagementReducer } from '../views/process-management';

import {reducer as layout2Reducer} from '../common/layout/_redux/layoutRedux'
import {reducer as tableReducer} from '../../src/common/tables/_redux/tableRedux'


import {reducer as basicInformationReducer} from '../views/basic-information/_redux/baseInfoRedux';
import LogsReducer from '../views/logs/LogReducesrs';

import BIReducer from '../views/BI/BIReducer';
import {reducer as inboxReducer} from "../views/inbox/_redux/inboxRedux";
import {reducer as authReducer} from '../../src/views/auth/_redux/authRedux'

import {reducer as ServiceDeskReducer} from "../views/service-desk/_redux/serviceDeskRedux";
import {reducer as SupervisorReducer} from "../views/supervisor/_redux/supervisorRedux";
import {reducer as CustomSettingsReducer} from "../views/customSettings/_redux/customSettingsRedux";
import {reducer as MenuReducer} from "../views/menu/_redux/menuRedux";
import {reducer as WorkDeskReducer} from "../views/work-desk/_redux/workDeskRedux";

export const rootReducer = combineReducers({
	auth: authReducer,
	user: userReducer,
	i18n: i18nReducer,
	basicInformation: basicInformationReducer,
	processManagement: ProcessManagementReducer,
	systemManagement: systemManagementReducer,
	ServiceDesk:ServiceDeskReducer,
	BI:BIReducer,
	trello:trelloReducer,
	inbox:inboxReducer,
	layout2:layout2Reducer,
	table:tableReducer,
	logs:LogsReducer,
	supervisor:SupervisorReducer,
	customSettingsReducer:CustomSettingsReducer,
	menuReducer:MenuReducer,
	workDeskReducer:WorkDeskReducer
});

const store = createStore(
	rootReducer,
	applyMiddleware(thunk)
);

store.dispatch(requestUserInfo());

store.dispatch(i18nInit());

export default store;
