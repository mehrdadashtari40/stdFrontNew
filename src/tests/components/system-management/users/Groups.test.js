import React from 'react';
import { shallow } from 'enzyme';
import { Provider } from 'react-redux';
import store from '../../../../store/configureStore';
import configureStore from 'redux-mock-store';
import * as selectActions from '../../../../views/system-management/SystemManagementActions';
import Groups from '../../../../views/system-management/components/users/Groups';

const mockGroupData = {
	grp_status: 'ACTIVE',
	grp_tasks: 2,
	grp_title: 'AUTO_آرشیو نامه ها',
	grp_uid: '7372551305a59d8f8c73f52010019622',
	grp_users: 6,
};

jest.mock('react-i18next', () => ({
	withTranslation: () => Component => {
		Component.defaultProps = { ...Component.defaultProps, t: () => '' };
		return Component;
	},
}));

test('Make Snapshot file , Groups Component', () => {
	const wrapper = shallow(
		<Provider store={store}>
			<Groups />
		</Provider>
	)
		.dive({ context: { store } })
		.dive();
	expect(wrapper).toMatchSnapshot();
});

it('invokes `componentDidMount` when mounted', () => {
	jest.spyOn(Groups.prototype, 'componentDidMount');
	shallow(
		<Provider store={store}>
			<Groups />
		</Provider>
	)
		.dive({ context: { store } })
		.dive();
	expect(Groups.prototype.componentDidMount).toHaveBeenCalled();
	Groups.prototype.componentDidMount.mockRestore();
});

test('handelIndex Function  ', () => {
	const wrapper = shallow(
		<Provider store={store}>
			<Groups />
		</Provider>
	)
		.dive({ context: { store } })
		.dive();
	const instance = wrapper.instance();
	expect(instance.handelIndex).toBeDefined();
});

test('handelCreate Function  ', () => {
	const wrapper = shallow(
		<Provider store={store}>
			<Groups />
		</Provider>
	)
		.dive({ context: { store } })
		.dive();
	const instance = wrapper.instance();
	expect(instance.handelCreate).toBeDefined();
});

test('handelAssignUsers Function  ', () => {
	const wrapper = shallow(
		<Provider store={store}>
			<Groups />
		</Provider>
	)
		.dive({ context: { store } })
		.dive();
	const instance = wrapper.instance();
	expect(instance.handelAssignUsers).toBeDefined();
});

test('handelShow Function  ', () => {
	const wrapper = shallow(
		<Provider store={store}>
			<Groups />
		</Provider>
	)
		.dive({ context: { store } })
		.dive();
	const instance = wrapper.instance();
	expect(instance.handelShow).toBeDefined();
});

test('SetData Function  ', () => {
	const wrapper = shallow(
		<Provider store={store}>
			<Groups />
		</Provider>
	)
		.dive({ context: { store } })
		.dive();
	const instance = wrapper.instance();
	expect(instance.SetData).toBeDefined();
});

test('enableFields Function  ', () => {
	const wrapper = shallow(
		<Provider store={store}>
			<Groups />
		</Provider>
	)
		.dive({ context: { store } })
		.dive();
	const instance = wrapper.instance();
	expect(instance.enableFields).toBeDefined();
});

test('disableFields Function  ', () => {
	const wrapper = shallow(
		<Provider store={store}>
			<Groups />
		</Provider>
	)
		.dive({ context: { store } })
		.dive();
	const instance = wrapper.instance();
	expect(instance.disableFields).toBeDefined();
});

test('assignUser Function  ', () => {
	const wrapper = shallow(
		<Provider store={store}>
			<Groups />
		</Provider>
	)
		.dive({ context: { store } })
		.dive();
	const instance = wrapper.instance();
	expect(instance.assignUser).toBeDefined();
});

test('unAssignUser Function  ', () => {
	const wrapper = shallow(
		<Provider store={store}>
			<Groups />
		</Provider>
	)
		.dive({ context: { store } })
		.dive();
	const instance = wrapper.instance();
	expect(instance.unAssignUser).toBeDefined();
});

test('setGroupsMembers Function  ', () => {
	const wrapper = shallow(
		<Provider store={store}>
			<Groups />
		</Provider>
	)
		.dive({ context: { store } })
		.dive();
	const instance = wrapper.instance();
	expect(instance.setGroupsMembers).toBeDefined();
});

test('handelDelete Function  ', () => {
	const wrapper = shallow(
		<Provider store={store}>
			<Groups />
		</Provider>
	)
		.dive({ context: { store } })
		.dive();
	const instance = wrapper.instance();
	expect(instance.handelDelete).toBeDefined();
});

test('getAvailableUsers Function  ', () => {
	const wrapper = shallow(
		<Provider store={store}>
			<Groups />
		</Provider>
	)
		.dive({ context: { store } })
		.dive();
	const instance = wrapper.instance();
	expect(instance.getAvailableUsers).toBeDefined();
});

test('closeAssignUserModal Function  ', () => {
	const wrapper = shallow(
		<Provider store={store}>
			<Groups />
		</Provider>
	)
		.dive({ context: { store } })
		.dive();
	const instance = wrapper.instance();
	expect(instance.closeAssignUserModal).toBeDefined();
});

describe('actions', () => {
	it('should create an action to loadGroups', () => {
		const data = mockGroupData;
		const expectedAction = {
			type: selectActions.LOAD_GROUPS,
			Groups: data,
		};
		expect(selectActions.setGroups(data)).toEqual(expectedAction);
	});
});
