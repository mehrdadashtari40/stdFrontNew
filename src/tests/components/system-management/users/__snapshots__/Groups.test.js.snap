// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Make Snapshot file , Groups Component 1`] = `
<div
  id="content"
>
  <div
    className="row"
  >
    <BigBreadcrumbs
      className="col-xs-12 col-sm-7 col-md-7 col-lg-4"
      icon="fa fa-fw fa-table"
      items={
        Array [
          "کاربران",
          "گروه",
        ]
      }
    />
  </div>
  <WidgetGrid>
    <div
      className="row"
    >
      <article
        className="col-sm-12"
      >
        <JarvisWidget
          collapsed={false}
          color={false}
          colorbutton={false}
          custombutton={false}
          deletebutton={false}
          editbutton={false}
          fullscreenbutton={true}
          hidden={false}
          load={false}
          refresh={false}
          sortable={true}
          togglebutton={false}
        >
          <header>
            <span
              className="widget-icon"
            >
               
              <i
                className="fa fa-bars"
              />
               
            </span>
            <h2>
               
               
            </h2>
            <div
              className="widget-toolbar"
            >
              <a
                className="btn btn-outline-success little_btn"
                onClick={[Function]}
              >
                 
                <i
                  className="fa fa-plus"
                />
                <span />
              </a>
            </div>
          </header>
          <div
            className="tab-content"
            id="myTabContent1"
          >
            <div
              className="tab-pane fade in active"
            >
              <Datatable
                className="table table-striped table-bordered table-hover"
                deleteAct={[Function]}
                filter={true}
                options={
                  Object {
                    "columns": Array [
                      Object {
                        "data": "grp_title",
                      },
                      Object {
                        "data": "grp_status",
                        "render": [Function],
                      },
                      Object {
                        "data": "grp_users",
                      },
                      Object {
                        "data": "grp_tasks",
                      },
                    ],
                    "data": Array [],
                  }
                }
                paginationLength={true}
                showAct={[Function]}
                width="100%"
              >
                <thead>
                  <tr>
                    <th
                      className="hasinput"
                      style={
                        Object {
                          "width": "18%",
                        }
                      }
                    >
                      <input
                        className="form-control "
                        id="dateselect_filter"
                        placeholder=""
                        type="text"
                      />
                    </th>
                    <th
                      className="hasinput"
                      style={
                        Object {
                          "width": "4%",
                        }
                      }
                    >
                      <input
                        className="form-control "
                        id="dateselect_filter"
                        placeholder=""
                        type="text"
                      />
                    </th>
                    <th
                      className="hasinput"
                      style={
                        Object {
                          "width": "4%",
                        }
                      }
                    >
                      <input
                        className="form-control "
                        id="dateselect_filter"
                        placeholder=""
                        type="text"
                      />
                    </th>
                    <th
                      className="hasinput"
                      style={
                        Object {
                          "width": "4%",
                        }
                      }
                    >
                      <input
                        className="form-control "
                        id="dateselect_filter"
                        placeholder=""
                        type="text"
                      />
                    </th>
                  </tr>
                  <tr>
                    <th
                      data-hide="phone"
                    >
                      <i
                        className="fa fa-fw fa-user text-muted hidden-md hidden-sm hidden-xs"
                      />
                    </th>
                    <th
                      data-hide="phone"
                    >
                      <i
                        className="fa fa-fw fa-star text-muted hidden-md hidden-sm hidden-xs"
                      />
                    </th>
                    <th
                      data-hide="phone,tablet"
                    >
                      <i
                        className="fa fa-fw fa-users txt-color-blue hidden-md hidden-sm hidden-xs"
                      />
                    </th>
                    <th
                      data-hide="phone,tablet"
                    >
                      <i
                        className="fa fa-fw fa-tasks txt-color-blue hidden-md hidden-sm hidden-xs"
                      />
                    </th>
                  </tr>
                </thead>
              </Datatable>
            </div>
          </div>
        </JarvisWidget>
      </article>
    </div>
  </WidgetGrid>
  <div
    className="modal fade"
    id="GroupBox"
    role="dialog"
  >
    <div
      className="modal-dialog"
    >
      <div
        className="modal-content"
      >
        <div
          className="modal-header"
        >
          <button
            className="close"
            data-dismiss="modal"
            type="button"
          >
            ×
          </button>
          <h4
            className="modal-title"
          >
             
          </h4>
        </div>
        <div
          className="modal-body"
        >
          <GroupForm
            Data={Array []}
            Disable={[Function]}
            Edit={false}
            Enable={[Function]}
            Refresh={[Function]}
            getUsers={false}
            setGroupsMembers={[Function]}
            t={[Function]}
          />
        </div>
      </div>
    </div>
  </div>
  <div
    className="modal fade"
    id="GroupMembers"
    role="dialog"
  >
    <div
      className="modal-dialog"
    >
      <div
        className="modal-content"
      >
        <div
          className="modal-header"
        >
          <button
            className="close"
            data-dismiss="modal"
            type="button"
          >
            ×
          </button>
          <h4
            className="modal-title"
          >
             
          </h4>
        </div>
        <div
          className="modal_option_bar padd_for_modal_optn"
        >
          <a
            className="btn btn-primary"
            onClick={[Function]}
          >
            <i
              className="fa fa-users"
            />
          </a>
        </div>
        <div>
          <Datatable
            className="table table-striped table-bordered table-hover"
            options={
              Object {
                "columns": Array [
                  Object {
                    "data": "usr_username",
                  },
                  Object {
                    "data": null,
                    "render": [Function],
                  },
                  Object {
                    "data": "usr_status",
                    "render": [Function],
                  },
                ],
                "data": Array [],
              }
            }
            paginationLength={true}
            width="100%"
          >
            <thead>
              <tr>
                <th
                  data-hide="phone"
                >
                  <i
                    className="fa fa-fw fa-user text-muted hidden-md hidden-sm hidden-xs"
                  />
                </th>
                <th
                  data-class="expand"
                >
                  <i
                    className="fa fa-fw fa-user text-muted hidden-md hidden-sm hidden-xs"
                  />
                </th>
                <th
                  data-hide="phone"
                >
                  <i
                    className="fa fa-fw fa-star text-muted hidden-md hidden-sm hidden-xs"
                  />
                  Status
                </th>
              </tr>
            </thead>
          </Datatable>
        </div>
        <div
          className="modal-footer"
        >
          <button
            className="btn btn-danger pull-right"
            data-dismiss="modal"
            type="button"
          >
            <i
              className="fa fa-close"
            />
             
          </button>
        </div>
      </div>
    </div>
  </div>
  <div
    className="modal fade"
    id="AssignUsers"
    role="dialog"
  >
    <div
      className="modal-dialog modal-lg"
    >
      <div
        className="modal-content"
      >
        <div
          className="modal-header"
        >
          <button
            className="close"
            onClick={[Function]}
            type="button"
          >
            ×
          </button>
          <h4
            className="modal-title"
          />
        </div>
        <div />
        <div
          className="modal-footer"
        >
          <button
            className="btn btn-default"
            data-dismiss="modal"
            onClick={[Function]}
            type="button btn btn-success"
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;
